### PR TITLE
Edit command

### PR DESCRIPTION
**Pull request by owen-yap**
- Added edit command
- Made changes to AddressBook, ReadOnlyAddressBook and Model so that they can process Doctors

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Doctor.java [hunk 1]
```diff
@@ -29,4 +30,67 @@ public Specialty getSpecialty() {
...
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getName())
+                .append("; Phone: ")
+                .append(getPhone())
+                .append("; Email: ")
+                .append(getEmail())
+                .append("; Specialty: ")
```
Comment (vigonometry) [inline src/main/java/seedu/address/model/person/Doctor.java:84]:
```suggestion
.append("; Speciality: ")
```
Standardise spelling across Docedex

File: src/main/java/seedu/address/logic/commands/EditDoctorCommand.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    public static final String COMMAND_WORD = "edit-doc";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the doctor identified "
+            + "by the index number used in the displayed doctors list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_PHONE + "PHONE "
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_SPECIALITY + "SPECIALITY "
```
Comment (vigonometry) [inline src/main/java/seedu/address/logic/commands/EditDoctorCommand.java:45]:
```suggestion
+ PREFIX_SPECIALTY + "SPECIALTY"
```

Standardise spelling across Docedex and use US English standard.

**Review by vigonometry — CHANGES_REQUESTED** [review]
Request updates on PREFIX + help string to match class names and US English standard.

**Comment by owen-yap** [comment]
Made a few changes to Add Command, changed the way doctors are added so that it is added under a separate list then persons. This will allow the edit command to work properly. 

@vigonometry

**Review by vigonometry — APPROVED** [review]

**Review by vigonometry — APPROVED** [review]
LGTM

**Pull request merged**